ament_cmake_gmock/ament_cmake_gmock-extras.cmake

```diff
@@ -71,6 +68,14 @@ macro(_ament_cmake_gmock_find_gmock)
 
         set(GMOCK_FROM_SOURCE_LIBRARIES "gmock" CACHE INTERNAL "")
         set(GMOCK_FROM_SOURCE_MAIN_LIBRARIES "gmock_main" CACHE INTERNAL "")
+      else()
+        # try to find and use gmock from GTest
+        find_package(GTest QUIET)
+        if(GTest_FOUND)
+          set(GMOCK_FOUND TRUE)
+          set(GMOCK_LIBRARIES GTest::gmock)
+          set(GMOCK_MAIN_LIBRARIES GTest::gmock_main)
+        endif()
```

Comment:
This looks a little strange to me. Mind explaining when we can assume `GTest` will provide targets for `gmock` prefixed with `GTest::`?

Reply:
Because it's in the documentation? :)
https://cmake.org/cmake/help/latest/module/FindGTest.html#imported-targets